lib/pure/times.nim

```diff
@@ -1245,7 +1245,9 @@ proc parse*(value, layout: string, zone: Timezone = local()): DateTime =
       result = initDateTime(zone.zoneInfoFromTz(result.toAdjTime), zone)
     else:
       # Otherwise convert to `zone`
-      result = result.toTime.inZone(zone)
+      result = result.toTime.inZone(zone)   
+  elif ptDate in found_tokens:
+    result.timezone = zone
```

Comment:
This isn't enough sadly. Due to how offset changes (e.g because of DST) works, not all combinations of date, time & timezone are valid. For example, if the clock moves forward one hour at `2017/03/12T02:00`, then all `DateTime`'s between `2017/03/12T02:00` and `2017/03/12T03:00` are invalid in that timezone. So if e.g `2017/03/12T02:30` is parsed it needs to be normalized to `2017/03/12T03:30`. This is the reasons for this line above:

           result = initDateTime(zone.zoneInfoFromTz(result.toAdjTime), zone)

Because of this I don't think it's possible to avoid dealing with timezones when parsing a `DateTime`. If you want speed, you can use send `utc()` as the `zone` argument which will be quick (in fact, it's possible to optimize that specific case a bit more).

Reply:
I don't think I have made changes that affect date&time pairs behavior, only when there is a date without time I putting dummy local timezone.

Reply:
The problem is that there is no such thing as a `DateTime` without time. You're implementation will use the default time of 00:00:00, which might not be valid depending on `zone`. F.ex in Brazil DST always starts and ends at 00:00:00, which will cause an invalid `DateTime`.

To do this properly, there must be additional types (`NaiveDateTime` for `DateTime` without timezone, `Date` for `DateTime` without time and timezone). Some languages use a single type for all cases, but that ends up being a confusing mess.

Like I said, to maximize speed the best option is currently (since we don't have the additional types yet) to use the UTC timezone, which we can optimize for. UTC doesn't require any normalization since it never changes.